Common/Utils/include/CommonUtils/BoostSerializer.h

```diff
@@ -33,61 +41,46 @@ namespace utils
 {
 namespace check
 {
-///A set of classes and struct to be sure the serialised object is either trivial or implementing custom serialize
-template <class Type, class Archive, typename = typename std::enable_if<std::is_class<Type>::value>::type>
-class is_boost_serializable
-{
- private:
-  struct TypeOverloader {
-    void serialize(Archive& ar, const unsigned int version) {}
-  };
-  struct TypeExt : public Type, public TypeOverloader {
-  };
-  template <typename T, T t>
-  class DeductionHelper
-  {
-  };
+// template <class Type, class Archive, typename = typename std::enable_if<std::is_class<Type>::value>::type>
+template <typename...>
+using void_t = void;
+
+template <typename Type, typename Archive = boost::archive::binary_oarchive, typename = void_t<>>
+struct is_boost_serializable_base : std::false_type {
+};
+
+template <class Type, typename Archive>
+struct is_boost_serializable_base<Type, Archive,
+                                  void_t<decltype(std::declval<Type &>().serialize(std::declval<Archive &>(), 0))>>
+  : std::true_type {
+};
+
+template <typename Type, typename Archive>
+struct is_boost_serializable_base<Type, Archive,
+                                  typename std::enable_if<boost::serialization::is_bitwise_serializable<Type>::value>::type>
+  : std::true_type {
+};
```

Comment:
these type traits should probably go into `Framework/Core/include/Framework/TypeTraits.h`